kolibri/plugins/learn/assets/src/state/actions/recommended.js

```diff
@@ -0,0 +1,280 @@
+import {
+  ContentNodeResource,
+  ContentNodeProgressResource,
+  SessionResource,
+  UserExamResource,
+  ExamLogResource,
+  ExamAttemptLogResource,
+} from 'kolibri.resources';
+
+import {
+  currentUserId,
+  isFacilityUser,
+  getCurrentChannelId,
+  getCurrentChannelObject,
+} from 'kolibri.coreVue.vuex.getters';
+
+import {
+  samePageCheckGenerator,
+  setChannelInfo,
+  handleApiError,
+} from 'kolibri.coreVue.vuex.actions';
+
+import { PageNames } from '../../constants';
+import { _contentState } from './main';
+
+import ConditionalPromise from 'kolibri.lib.conditionalPromise';
+import uniqBy from 'lodash/uniqBy';
+
+function _fetchPopular(channelId) {
+  const channelPayload = { channel_id: channelId };
+  const popularPayload = { popular: 'true' };
+  return ContentNodeResource.getCollection(channelPayload, popularPayload).fetch();
+}
+
+function _fetchNextSteps(channelId, state) {
+  const nextStepsPayload = { next_steps: currentUserId(state) };
+  const channelPayload = { channel_id: channelId };
+
+  if (isFacilityUser(state)) {
+    return ContentNodeResource.getCollection(channelPayload, nextStepsPayload).fetch();
+  }
+  return Promise.resolve([]);
+}
+
+function _fetchResume(channelId, state) {
+  const resumePayload = { resume: currentUserId(state) };
+  const channelPayload = { channel_id: channelId };
+
+  // all you really need is user_id. Should we check in parent?
+  if (isFacilityUser(state)) {
+    return ContentNodeResource.getCollection(channelPayload, resumePayload).fetch();
+  }
+  return Promise.resolve([]);
+}
+
+function _fetchOverview(channelId, cursor) {
```

Comment:
@rtibbles I have no experience with database cursors. Is there any way I can make this give me 40 pieces of content? Random would be best, but any 40 would do.

I know I can use `ContentNodeResource.getCollection(channelPayload, resumePayload).next` after a `fetch`, but I'd have to do that 4 times to get 40 items, per the limits that @khangmach, @indirectlylit, @christianmemije, and I came up with. Please help.